include/MiniFB_ex_cpp.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#if defined(__cplusplus)
+
+#include <functional>
+#include "MiniFB_ex.h"
+
+template <class T>
+void mfb_active_callback(T *obj, void (T::*method)(eBool));
+
+template <class T>
+void mfb_resize_callback(T *obj, void (T::*method)(int, int));
+
+template <class T>
+void mfb_keyboard_callback(T *obj, void (T::*method)(eKey, eKeyMod, eBool));
+
+template <class T>
+void mfb_char_input_callback(T *obj, void (T::*method)(unsigned int));
+
+template <class T>
+void mfb_mouse_button_callback(T *obj, void (T::*method)(eMouseButton, eKeyMod, eBool));
+
+template <class T>
+void mfb_mouse_move_callback(T *obj, void (T::*method)(int, int));
+
+template <class T>
+void mfb_mouse_scroll_callback(T *obj, void (T::*method)(eKeyMod, float, float));
+
+//-------------------------------------
+// To avoid clumsy hands
+//-------------------------------------
+class Stub {
+    template <class T>
+    friend void mfb_active_callback(T *obj, void (T::*method)(eBool));
+    template <class T>
+    friend void mfb_resize_callback(T *obj, void (T::*method)(int, int));
+    template <class T>
+    friend void mfb_mouse_button_callback(T *obj, void (T::*method)(eMouseButton, eKeyMod, eBool));
+    template <class T>
+    friend void mfb_keyboard_callback(T *obj, void (T::*method)(eKey, eKeyMod, eBool));
+    template <class T>
+    friend void mfb_char_input_callback(T *obj, void (T::*method)(unsigned int));
+    template <class T>
+    friend void mfb_mouse_button_callback(T *obj, void (T::*method)(eMouseButton, eKeyMod, eBool));
+    template <class T>
+    friend void mfb_mouse_move_callback(T *obj, void (T::*method)(int, int));
+    template <class T>
+    friend void mfb_mouse_scroll_callback(T *obj, void (T::*method)(eKeyMod, float, float));
+
+    static void active_stub(eBool isActive);
+    static void resize_stub(int width, int height);
+    static void keyboard_stub(eKey key, eKeyMod mod, eBool isPressed);
+    static void char_input_stub(unsigned int);
+    static void mouse_btn_stub(eMouseButton button, eKeyMod mod, eBool isPressed);
+    static void mouse_move_stub(int x, int y);
+    static void scroll_stub(eKeyMod mod, float deltaX, float deltaY);
+
+    static std::function<void(eBool)>                        m_active;
+    static std::function<void(int, int)>                     m_resize;
+    static std::function<void(eKey, eKeyMod, eBool)>         m_keyboard;

```

Comment:
I would prefer the naming of `eKey/eKeyMod` to not have an `e` in-front of them if possible

Reply:
ok, I usually do that to easily differentiate enums